File: app/src/main/java/ru/santaev/clipboardtranslator/db/entity/Translation.java
package ru.santaev.clipboardtranslator.db.entity;


import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

import static ru.santaev.clipboardtranslator.db.entity.TranslationContract.*;

@Entity(tableName = TranslationContract.TABLE_NAME)
public class Translation {

    @PrimaryKey(autoGenerate = true)
    private long id;

    @ColumnInfo(name = LANG_SOURCE)
    private String langSource;

    @ColumnInfo(name = LANG_TARGET)
    private String langTarget;

    @ColumnInfo(name = TEXT_SOURCE)
    private String textSource;

    @ColumnInfo(name = TEXT_TARGET)
    private String textTarget;

    public Translation(String langSource, String langTarget, String textSource, String textTarget) {
        this.langSource = langSource;
        this.langTarget = langTarget;
        this.textSource = textSource;
        this.textTarget = textTarget;
    }

    public void setId(long id) {
        this.id = id;
    }

    public void setLangSource(String langSource) {
        this.langSource = langSource;
    }

    public void setLangTarget(String langTarget) {
        this.langTarget = langTarget;
    }

    public void setTextSource(String textSource) {
        this.textSource = textSource;
    }

    public void setTextTarget(String textTarget) {
        this.textTarget = textTarget;
    }

    public long getId() {
        return id;
    }

    public String getLangSource() {
        return langSource;
    }

    public String getLangTarget() {
        return langTarget;
    }

    public String getTextSource() {
        return textSource;
    }

    public String getTextTarget() {
        return textTarget;
    }
}

File: app/src/main/java/ru/santaev/clipboardtranslator/db/entity/Translation.kt
package ru.santaev.clipboardtranslator.db.entity


import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

import ru.santaev.clipboardtranslator.db.entity.TranslationContract.LANG_SOURCE
import ru.santaev.clipboardtranslator.db.entity.TranslationContract.LANG_TARGET
import ru.santaev.clipboardtranslator.db.entity.TranslationContract.TEXT_SOURCE
import ru.santaev.clipboardtranslator.db.entity.TranslationContract.TEXT_TARGET

@Entity(tableName = TranslationContract.TABLE_NAME)
class Translation(
        @PrimaryKey(autoGenerate = true) var id: Long = 0,
        @ColumnInfo(name = LANG_SOURCE) var langSource: String?,
        @ColumnInfo(name = LANG_TARGET) var langTarget: String?,
        @ColumnInfo(name = TEXT_SOURCE) var textSource: String?,
        @ColumnInfo(name = TEXT_TARGET) var textTarget: String?
) {

    @Ignore
    constructor(
            langSource: String,
            langTarget: String,
            textSource: String,
            textTarget: String
    ) : this(0, langSource, langTarget, textSource, textTarget)
}

